system/lib/wasmfs/wasmfs.cpp

```diff
@@ -103,6 +104,43 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+long __syscall_fstat64(long fd, long buf) {
+  // Release FileTable lock after accessing desired open file.
+  auto currentOpenFile = [&fd] {
+    auto fileTable = FileTable::get();
+    return fileTable[fd]->get();
+  }();
+
+  auto file = currentOpenFile.getFile();
+  auto fileHandle = file->get();
+
+  struct stat* buffer = (struct stat*)buf;
+  buffer->st_dev = 1; // ID of device containing file: Hardcode 1 for now, no meaning at the
+  // moment for Emscripten.
+  buffer->st_mode = fileHandle.mode();
+  buffer->st_ino = fd;
+  // The number of hard links is 1 since they are unsupported.
+  buffer->st_nlink = 1;
+  buffer->st_uid = 0;
+  buffer->st_gid = 0;
+  buffer->st_rdev = 1; // Device ID (if special file) No meaning right now for Emscripten.
+  if (file->is<Directory>()) {
+    buffer->st_size = 4096;
```

Comment:
It would be good to add comments explaining that these values are chosen for compatibility.